README.md

```diff
@@ -17,59 +21,495 @@ kubectl apply -k 'github.com/howardjohn/service-apis-demo'
+
+```bash
+$ kubectl apply -f traffic-split/traffic-split-httproute-1.yaml
+httproute.networking.x-k8s.io/foo-store-route created
+
+$ kubectl describe httproute foo-store-route
+Name:         foo-store-route
+Namespace:    gateway-demo
+Labels:       <none>
+API Version:  networking.x-k8s.io/v1alpha1
+Kind:         HTTPRoute
+Spec:
+  Hosts:
+    Hostnames:
+      foo.com
+    Rules:
+      Action:
+        Forward To:
+          Target Ref:
+            Group:     core
+            Name:      foo-store-svc-v1
+            Resource:  services
+          Weight:      100
+          Target Ref:
+            Group:     core
+            Name:      foo-store-svc-v2
+            Resource:  services
+          Weight:      0
+      Match:
+        Path:  /store
+```
+
+14. In a separate shell, start a continuous curl to demonstrate the traffic split. At this point in time all responses should be coming from v1 of the application.
+
+```bash
+$ while true; do curl http://$INGRESS_HOST:$INGRESS_PORT/store -H "host: foo.com" -s | jq '[.pod_name, .timestamp]' -cr; sleep 1; done
+
+["foo-store-v1-66df6b4f69-v6m9k","2020-08-16T19:53:22"]
+["foo-store-v1-66df6b4f69-v6m9k","2020-08-16T19:53:23"]
+["foo-store-v1-66df6b4f69-v6m9k","2020-08-16T19:53:25"]
+...
+```
+
+15. Now apply the 2nd stage of the traffic split to start gracefully shifting 20% of the traffic to v2.
+
+```bash
+$ kubectl apply -f traffic-split/traffic-split-httproute-2.yaml
+httproute.networking.x-k8s.io/foo-store-route configured
+```
+
+In the other shell, observe the change in traffic behavior. You should now see that roughly 20% of the responses are coming from v2.
+
+```bash
+["foo-store-v2-77fbd98c7f-5qk6c","2020-08-16T19:55:50"]
+["foo-store-v1-66df6b4f69-v6m9k","2020-08-16T19:55:51"]
+["foo-store-v1-66df6b4f69-v6m9k","2020-08-16T19:55:53"]
+["foo-store-v2-77fbd98c7f-5qk6c","2020-08-16T19:55:54"]
+["foo-store-v1-66df6b4f69-v6m9k","2020-08-16T19:55:55"]
+["foo-store-v1-66df6b4f69-v6m9k","2020-08-16T19:55:56"]
+["foo-store-v1-66df6b4f69-v6m9k","2020-08-16T19:55:58"]
+...
+```
+
+16. Now that we've observed that v2 responses are performing as expected without errors, we can apply the final step of the traffic split and shift all traffic over to v2.
+
+```bash
+$ kubectl apply -f traffic-split/traffic-split-httproute-3.yaml
+httproute.networking.x-k8s.io/foo-store-route configured
+```
+
+Again, validate the responses in the other shell to ensure that v2 is now handilng all of the traffic.
+
+```bash
+...
+["foo-store-v2-77fbd98c7f-5qk6c","2020-08-16T19:58:05"]
+["foo-store-v2-77fbd98c7f-5qk6c","2020-08-16T19:58:07"]
+["foo-store-v2-77fbd98c7f-5qk6c","2020-08-16T19:58:08"]
+["foo-store-v2-77fbd98c7f-5qk6c","2020-08-16T19:58:09"]
+...
+```
+
+
+
+### Clean up
+
+To delete everything that was created in the `gateway-demo` Namespace, issue the following command.
+
+```bash
+$ kubectl delete -R -f ./service-apis-demo -n gateway-demo
```

Comment:
Everything else assumes we are in `./service-apis-demo` already. I also got a bunch of errors doing this when I replaced with `.`:

```
the namespace from the provided object "istio-system" does not match the namespace "gateway-demo". You must pass '--namespace=istio-system' to perform this operation.
the namespace from the provided object "istio-system" does not match the namespace "gateway-demo". You must pass '--namespace=istio-system' to perform this operation.
the namespace from the provided object "istio-system" does not match the namespace "gateway-demo". You must pass '--namespace=istio-system' to perform this operation.
the namespace from the provided object "istio-system" does not match the namespace "gateway-demo". You must pass '--namespace=istio-system' to perform this operation.
Error from server (NotFound): error when deleting "single-service-gateway/httproute.yaml": httproutes.networking.x-k8s.io "simple-route" not found
the namespace from the provided object "istio-system" does not match the namespace "gateway-demo". You must pass '--namespace=istio-system' to perform this operation.
the namespace from the provided object "service-apis-demo" does not match the namespace "gateway-demo". You must pass '--namespace=service-apis-demo' to perform this operation.
Error from server (NotFound): error when deleting "tls/example.yaml": gatewayclasses.networking.x-k8s.io "istio" not found
the namespace from the provided object "istio-system" does not match the namespace "gateway-demo". You must pass '--namespace=istio-system' to perform this operation.
the namespace from the provided object "service-apis-demo" does not match the namespace "gateway-demo". You must pass '--namespace=service-apis-demo' to perform this operation.
the namespace from the provided object "service-apis-demo" does not match the namespace "gateway-demo". You must pass '--namespace=service-apis-demo' to perform this operation.
Error from server (NotFound): error when deleting "tls/httpbin.yaml": namespaces "service-apis-demo" not found
the namespace from the provided object "service-apis-demo" does not match the namespace "gateway-demo". You must pass '--namespace=service-apis-demo' to perform this operation.
the namespace from the provided object "service-apis-demo" does not match the namespace "gateway-demo". You must pass '--namespace=service-apis-demo' to perform this operation.
the namespace from the provided object "service-apis-demo" does not match the namespace "gateway-demo". You must pass '--namespace=service-apis-demo' to perform this operation.
Error from server (NotFound): error when deleting "traffic-split/traffic-split-httproute-2.yaml": httproutes.networking.x-k8s.io "foo-store-route" not found
Error from server (NotFound): error when deleting "traffic-split/traffic-split-httproute-3.yaml": httproutes.networking.x-k8s.io "foo-store-route" not found
[unable to decode "kustomization.yml": Object 'Kind' is missing in '{"resources":["deploy.yaml"]}', unable to decode "tls/kustomization.yaml": Object 'Kind' is missing in '{"resources":["httpbin.yaml","certificate.yaml","example.yaml","example-ingress.yaml"]}']
```